[motion-path] Inset shape tests with border-radius are failing

nt1m opened the pull request:
#### 07d04d5b029f210076393cb6535098b286139993
<pre>
[motion-path] Inset shape tests with border-radius are failing
<a href="https://bugs.webkit.org/show_bug.cgi?id=261437">https://bugs.webkit.org/show_bug.cgi?id=261437</a>
rdar://115316728

Reviewed by Darin Adler.

By default, `addRoundedRect` will use `PathRoundedRect::Strategy::PreferNative`, which will change the starting position of the path as soon as border-radius is found.

While the starting position doesn&apos;t affect clip-path rendering, it does affect motion path. Switch to `PathRoundedRect::Strategy::PreferBezier` so `addRoundedRect` returns
a path with a consistent starting point (top left corner) regardless of border-radius.

This affects all inset shapes (inset, rect, xywh).

* LayoutTests/TestExpectations:
* LayoutTests/css3/masking/clip-path-border-radius-border-box-000.html:
* LayoutTests/css3/masking/clip-path-border-radius-content-box-001.html:
* LayoutTests/css3/masking/clip-path-border-radius-fill-box-001.html:
* LayoutTests/css3/masking/clip-path-border-radius-padding-box-001.html:
* LayoutTests/css3/masking/clip-path-border-radius-stroke-box-000.html:
* LayoutTests/css3/masking/clip-path-border-radius-view-box-000.html:
* LayoutTests/css3/masking/clip-path-inset.html:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/offset-path-shape-inset-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/offset-path-shape-rect-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/offset-path-shape-xywh-002.html:
* Source/WebCore/rendering/style/BasicShapes.cpp:
(WebCore::RoundedRectPathPolicy::createValueForKey):
* Source/WebCore/rendering/MotionPath.cpp:
(WebCore::MotionPath::motionPathDataForRenderer): Drive-by cleanup

Canonical link: <a href="https://commits.webkit.org/268192@main">https://commits.webkit.org/268192@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/53bc5b1eccdf07f9ac8592ca6ebb979015709118

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18964 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/19308 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/19915 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/20831 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17730 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/19152 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/22602 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/19434 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/19506 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/19189 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/19298 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/16506 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21719 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/16506 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/17282 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/23689 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/17546 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17454 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/21630 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18032 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/15292 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/17114 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/17116 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4512 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/21475 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17866 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler reviewed (approved):
This looks fine, but I see a lot of tests failing so it might be premature for me to approve.

shallawa commented on Source/WebCore/rendering/style/BasicShapes.cpp:
```diff
@@ -99,7 +99,7 @@ struct RoundedRectPathPolicy : public TinyLRUCachePolicy<FloatRoundedRect, Path>
     {
         Path path;
-        path.addRoundedRect(rect);
+        path.addRoundedRect(rect, PathRoundedRect::Strategy::PreferBezier);
```
How can this change affect the layout of the motion-path? What exactly does the native RoundedRect behave differently from the equivalent Bézier? Also does this css also require Bézier?
```
  offset-path: inset(50px round 0%);
```
or
```
  offset-path: inset(50%);
```

nt1m replied:
`offset-path: inset(50px round 0%);` and `offset-path: inset(50%);` are equivalent. 

But once you start using `offset-path: inset(50px round 1px);`, the starting position of the path for `PathRoundedRect::Strategy::PreferNative` goes from the top left to the bottom right. 

In the case of clip-path, it doesn't really matter because only the end shape matters, but for motion path, where the starting point of the path is matters (top left vs. bottom right)

nt1m committed:
[motion-path] Inset shape tests with border-radius are failing
https://bugs.webkit.org/show_bug.cgi?id=261437
rdar://115316728

Reviewed by Darin Adler.

By default, `addRoundedRect` will use `PathRoundedRect::Strategy::PreferNative`, which will change the starting position of the path as soon as border-radius is found.

While the starting position doesn't affect clip-path rendering, it does affect motion path. Switch to `PathRoundedRect::Strategy::PreferBezier` so `addRoundedRect` returns
a path with a consistent starting point (top left corner) regardless of border-radius.

This affects all inset shapes (inset, rect, xywh).

* LayoutTests/TestExpectations:
* LayoutTests/css3/masking/clip-path-border-radius-border-box-000.html:
* LayoutTests/css3/masking/clip-path-border-radius-content-box-001.html:
* LayoutTests/css3/masking/clip-path-border-radius-fill-box-001.html:
* LayoutTests/css3/masking/clip-path-border-radius-padding-box-001.html:
* LayoutTests/css3/masking/clip-path-border-radius-stroke-box-000.html:
* LayoutTests/css3/masking/clip-path-border-radius-view-box-000.html:
* LayoutTests/css3/masking/clip-path-inset.html:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/offset-path-shape-inset-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/offset-path-shape-rect-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/offset-path-shape-xywh-002.html:
* Source/WebCore/rendering/style/BasicShapes.cpp:
(WebCore::RoundedRectPathPolicy::createValueForKey):
* Source/WebCore/rendering/MotionPath.cpp:
(WebCore::MotionPath::motionPathDataForRenderer): Drive-by cleanup

Canonical link: https://commits.webkit.org/268192@main
